File: 체스판 다시 칠하기.cpp
#include <iostream>

int main()
{
	int n,m;
	std::cin >> n >> m;
	char board[51][51];
	
	for(int i=0; i<n; i++) {
		std::cin >> board[i];
	}
	
	int bw=0, wb=0, min=9999;
	for(int i=0; i<n-7; i++) {
		for(int j=0; j<m-7; j++) {
			bw = wb = 0;
			for(int k=i; k<i+8; k++) {
				for(int l=j; l<j+8; l++) {
					if(k%2==i%2) {
						if(l%2==j%2) {
							if(board[k][l] == 'B') {
								bw++;
							} else {
								wb++;
							}
						} else {
							if(board[k][l] == 'W') {
								bw++;
							} else {
								wb++;
							}
						}
					} else {
						if(l%2==j%2) {
							if(board[k][l] == 'W') {
								bw++;
							} else {
								wb++;
							}
						} else {
							if(board[k][l] == 'B') {
								bw++;
							} else {
								wb++;
							}
						}
					}
				}
			}
			if(min > wb) 
				min = wb;
			if(min > bw)
				min = bw;
		}
	} 
	std::cout << min;
	return 0;	
}

File: 항해 리부트/5일차/체스판 다시 칠하기.py
n, m = map(int, input().split())
board = [input() for _ in range(n)]
min_ = 9999
for i in range(m-7):
    for j in range(n-7):
        # 첫칸 색깔 
        min_black, min_white = 0, 0

        for k in range(8):
            for l in range(8):
                # 0 2 4 6 번째 줄, 0 2 4 6 번째 칸 and 1 3 5 7 번째 줄 1 3 5 7 번째 칸
                if k % 2 == 0 and l % 2 == 0 or k % 2 == 1 and l % 2 == 1:
                    if board[j+k][i+l] != "B":
                        min_black += 1
                    else:
                        min_white += 1
                else:
                    if board[j+k][i+l] != "W":
                        min_black += 1
                    else:
                        min_white += 1
        min_ = min(min_, min_black, min_white)
print(min_)
